src/Elm/Print.hs

```diff
@@ -206,11 +213,152 @@ elmEnumUniverse t@ElmType{..} = vsep
+        conName = pretty elmConstructorName
+
+        -- | Creates variables: @x1@ to @xN@, where N is the number of the constructor fields.
+        fields :: [Doc ann]
+        fields = take (length elmConstructorFields) $
+            map (pretty . mkText "x") [1..]
+
+        contents :: Doc ann
+        contents = "," <+> parens (dquotes "contents" <> comma <+> "E.list" <+> brackets fieldEncs)
+
+        -- | @encoderA x1@
+        fieldEncs :: Doc ann
+        fieldEncs = concatWith (surround ", ") $
+            zipWith (<+>) (map typeRefEncoder elmConstructorFields) fields
+
+        -- | Makes variable like: @x11@ etc.
+        mkText :: Text -> Int -> Text
+        mkText x i = x <> T.pack (show i)
+
+        vars :: Doc ann
+        vars =  concatWith (surround " ") fields
+
+
+aliasEncoderDoc :: ElmAlias -> Doc ann
+aliasEncoderDoc ElmAlias{..} =
+    encoderDef elmAliasName []
+    <> line
+    <> nest 4 (vsep $ (encoderName elmAliasName <+> "x" <+> equals <+> "E.object")
+            : fieldsEncode elmAliasFields)
+  where
+    fieldsEncode :: NonEmpty ElmRecordField -> [Doc ann]
+    fieldsEncode (fstR :| rest) =
+        lbracket <+> recordFieldDoc fstR
+      : map ((comma <+>) . recordFieldDoc) rest
+     ++ [rbracket]
+
+    recordFieldDoc :: ElmRecordField -> Doc ann
+    recordFieldDoc ElmRecordField{..} = parens $
+            dquotes (pretty elmRecordFieldName)
+         <> comma
+        <+> typeRefEncoder elmRecordFieldType
+        <+> "x." <> pretty elmRecordFieldName
+
+-- | The definition of the @encodeTYPENAME@ function.
+encoderDef
+    :: Text  -- ^ Type name
+    -> [Text] -- ^ List of type variables
+    -> Doc ann
+encoderDef typeName vars =
+    encoderName typeName <+> colon <+> pretty typeName <> typeVarsDoc <+> arrow <+> "Value"
+  where
+    typeVarsDoc :: Doc ann
+    typeVarsDoc = concatWith (surround " ") $ map pretty vars
+
+-- | Create the name of the encoder function.
+encoderName :: Text -> Doc ann
+encoderName typeName = "encode" <> pretty typeName
+
+-- | Converts the reference to the existing type to the corresponding encoder.
+typeRefEncoder :: TypeRef -> Doc ann
+typeRefEncoder (RefPrim elmPrim) = case elmPrim of
+    ElmUnit       -> "const E.null"
+    ElmNever      -> "never"
+    ElmBool       -> "E.bool"
+    ElmChar       -> parens "E.string << String.fromChar"
+    ElmInt        -> "E.int"
+    ElmFloat      -> "E.float"
+    ElmString     -> "E.string"
+    ElmMaybe t    -> parens $ "encodeMaybe" <+> typeRefEncoder t
+    ElmResult l r -> parens $ "encodeEither" <+> typeRefEncoder l <+> typeRefEncoder r
+    ElmPair a b   -> parens $ "encodePair" <+> typeRefEncoder a <+> typeRefEncoder b
+    ElmList l     -> "E.list" <+> typeRefEncoder l
+typeRefEncoder (RefCustom TypeName{..}) = "encode" <> pretty unTypeName
+
+encodeMaybe :: Doc ann
+encodeMaybe = vsep
+    [ "encodeMaybe : (a -> Value) -> Maybe a -> Value"
+    , "encodeMaybe enc = Maybe.withDefault E.null << Maybe.map enc"
```

Comment:
Yeah, it was expected that we will have some utility functions. I guess we will have some module like `ElmStreet` (or maybe better name) where we will put all such definitions and reexport them.